mobile/lib/services/auth.service.dart

```diff
@@ -95,4 +125,67 @@ class AuthService {
       rethrow;
     }
   }
+
+  Future<String?> setOpenApiServiceEndpoint() async {
+    final enable = _authRepository.getEndpointSwitchingFeature();
+    if (!enable) {
+      return null;
+    }
+
+    final wifiName = await _networkService.getWifiName();
+    final savedWifiName = _authRepository.getPreferredWifiName();
+    String? endpoint;
+
+    if (wifiName == savedWifiName) {
+      endpoint = await _setLocalConnection();
+    }
+
+    endpoint ??= await _setRemoteConnection();
+
+    return endpoint;
+  }
+
+  Future<String?> _setLocalConnection() async {
+    try {
+      final localEndpoint = _authRepository.getLocalEndpoint();
+      if (localEndpoint != null) {
+        await _apiService.resolveAndSetEndpoint(localEndpoint);
+        return localEndpoint;
+      }
+    } catch (error, stackTrace) {
+      _log.severe("Cannot set local endpoint", error, stackTrace);
+    }
+
+    return null;
+  }
+
+  Future<String?> _setRemoteConnection() async {
+    try {
+      final endpointList = _authRepository.getExternalEndpointList();
+      String? validUrl;
+
+      for (final endpoint in endpointList) {
+        // If endpoint or server is not online, try the next endpoint
+        try {
+          validUrl = await _apiService.resolveEndpoint(endpoint.url);
+        } on ApiException catch (error) {
+          _log.severe("Cannot resolve endpoint", error);
+          continue;
+        }
+
+        final isValid = await validateAuxilaryServerUrl(validUrl);
+        if (isValid) {
+          final abc = await _apiService.resolveAndSetEndpoint(validUrl);
+          print("[ABC] $abc");
```

Comment:
Should be a log or a debugPrint

Reply:
We should add the `avoid-print` lint

Reply:
Oh never mind it's already enabled. But also wrap debugPrint in `kIsDebug`.

Reply:
I am adding more tests and perform test debugging, forgot to remove this one, will do in a sec

Reply:
> Oh never mind it's already enabled. But also wrap `debugPrint` in `kIsDebug`.

TBH, we should change all the debugPrints to fine logs and add the isDebug flag in the ImmichLogger where we debugPrint all logs for development